resource/SQLiteDB.js

```diff
@@ -0,0 +1,49 @@
+// used for testing
+const sqliteDatabase = require('better-sqlite3');
+
+module.exports.createSQLiteDatabase = function(cb) {
+    // Please look at the README for debugging to the console
+    const Testdb = new sqliteDatabase('airlinesTest.db', {memory: true});
+    Testdb.prepare('CREATE TABLE `tbl_airports` (`code` char(3) NOT NULL,`name` varchar(45) NOT NULL,PRIMARY KEY (`code`),UNIQUE (`code`));').run();
+
+    Testdb.prepare('CREATE TABLE `tbl_flights` (`id` INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,`departure` char(3) NOT NULL,`destination` char(3) ' +
+        'NOT NULL,`departure_date` TEXT NOT NULL,`arrival_date` TEXT NOT NULL,`flight_number` int(11) NOT NULL,UNIQUE (`id`), FOREIGN KEY (`departure`) ' +
+        'REFERENCES `tbl_airports` (`code`), FOREIGN KEY (`destination`) REFERENCES `tbl_airports` (`code`),CONSTRAINT `departure!=destination` CHECK (`departure` != `destination`),' +
+        'CONSTRAINT `departure_date<destination_date` CHECK (`departure_date` < `arrival_date`));').run();
+    // create index for `destination` for tbl_flights
+    Testdb.prepare('CREATE UNIQUE INDEX destination_idx ON tbl_flights (destination);').run();
+    // create index for `departure` for tbl_flights
+    Testdb.prepare('CREATE UNIQUE INDEX departure_idx ON tbl_flights (departure);').run();
+
+    Testdb.prepare('CREATE TABLE `tbl_users` (`id` INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,`username` varchar(45),`password` varchar(255),`display_name` varchar(45) DEFAULT NULL,`role` int(11) ' +
+        'NOT NULL,`email` varchar(45),`phone` varchar(15),UNIQUE (`id`),CONSTRAINT `identifier_exists` CHECK ((`username` IS NOT NULL) OR (`email` IS NOT NULL) OR (`phone` IS NOT NULL)));').run();
+
+    Testdb.prepare('CREATE TABLE `tbl_tickets` (`flight` int(11) NOT NULL,`seat_row` int(11) NOT NULL,`seat` char(1) NOT NULL,`class` int(11) NOT NULL,`reserver` int(11) DEFAULT NULL,`price` int(11) ' +
+        'DEFAULT NULL,`reservation_timeout` datetime DEFAULT NULL,`booking_id` VARCHAR(45) DEFAULT NULL,PRIMARY KEY (`flight`,`seat_row`,`seat`),CONSTRAINT `flight` FOREIGN KEY (`flight`) REFERENCES `tbl_flights` (`id`),' +
+        'CONSTRAINT `reserver` FOREIGN KEY (`reserver`) REFERENCES `tbl_users` (`id`),CONSTRAINT `reserved_check` CHECK ( (`reserver` IS NULL AND `price` IS NULL AND `reservation_timeout` IS NULL)OR (`reserver` IS NOT NULL ' +
+        'AND `price` IS NOT NULL AND `reservation_timeout` IS NULL)OR (`reserver` IS NOT NULL AND `price` IS NULL AND `reservation_timeout` IS NOT NULL) ),CONSTRAINT `booking_id_check` CHECK ((`reserver` IS NULL AND `booking_id` IS NULL) ' +
+        'OR (`reserver` IS NOT NULL AND `booking_id` IS NOT NULL)));').run();
+    // create index for `reserver` for tbl_tickets
+    Testdb.prepare('CREATE UNIQUE INDEX reserver_idx ON tbl_tickets (reserver);').run();
+
+    Testdb.prepare('INSERT INTO `tbl_users` (`username`, `role`) VALUES ("user", "1");').run();
```

Comment:
For these `INSERT` statements, you're creating "prepared statements"; pass the values to be inserted in via the `run` method, even if they aren't actually user-provided data.

Reply:
I could not find any other function

Reply:
According to the documentation I read, the `run` method (which you call at the end of each of those lines) takes any prepared-statement parameters as its arguments. So this should be something like `prepare('INSERT INTO tbl_users (username, role) VALUES (?, ?)').run('user', 1)` (but check the documentation for the precise syntax).